app/models/user-work.ts

```diff
@@ -0,0 +1,304 @@
+ *
+ * @param tx - The database transaction
+ * @param serviceID - The ID of the service
+ * @returns The sum of ready and running work items for the service
+ */
+export async function getQueuedAndRunningCountForService(tx: Transaction, serviceID: string)
+  : Promise<number> {
+  const results = await tx(UserWork.table)
+    .sum({ readyCount: 'ready_count', runningCount: 'running_count' })
+    .where({ service_id: serviceID });
+
+  const totalItems = Number(results[0].readyCount) + Number(results[0].runningCount);
+
+  return totalItems;
+}
+
+/**
+ * Gets the next username that should have a work item worked for the given service ID
+ * SELECT username, SUM("u"."running_count") as s from user_work u WHERE username in
+ * (SELECT DISTINCT username FROM user_work u WHERE "u"."service_id" = 'SERVICE' AND "u"."ready_count" \> 0)
+ * GROUP BY username order by s, max(last_worked) asc LIMIT 1;
+
+ * @param tx - The database transaction
+ * @param serviceID - The ID of the service
+ * @returns The username that should have a work item worked next
+ */
+export async function getNextUsernameForWork(tx: Transaction, serviceID: string)
+  : Promise<string> {
+  const subquery = tx(UserWork.table)
+    .distinct('username')
+    .where('service_id', '=', serviceID)
+    .where('ready_count',  '>',  0);
+
+  const results = await tx(UserWork.table)
+    .select('username')
+    .max('last_worked as lw')
+    .sum('running_count as rc')
+    .whereIn('username', subquery)
+    .groupBy('username')
+    .orderBy('rc', 'asc')
+    .orderBy('lw', 'asc')
+    .first();
+
+  return results?.username;
+}
+
+/**
+ * Gets the next job to work on for the given username and service ID.
+ * select job_id from user_work where username = <username> and ready_count \> 0 order by last_worked asc limit 1
+ * @param tx - The database transaction
+ * @param serviceID - The ID of the service
+ * @param username - The username to choose a job to work on
+ * @returns The job ID that should have a work item worked next for the service
+ */
+export async function getNextJobIdForUsernameAndService(tx: Transaction, serviceID: string, username: string)
+  : Promise<string> {
+  const results = await tx(UserWork.table)
+    .select('job_id')
+    .where({ username, service_id: serviceID })
+    .where('ready_count', '>', 0)
+    .orderBy('is_async', 'asc')
+    .orderBy('last_worked', 'asc')
+    .first();
+
+  return results.job_id;
+}
+
+/**
+ * Deletes all of the rows for the given job from the user_work table.
+ * delete from user_work where job_id = $job_id
+ * @param tx - The database transaction
+ * @param jobID - The job ID
+ * @returns the number of rows deleted
+ */
+export async function deleteUserWorkForJob(tx: Transaction, jobID: string): Promise<number> {
+  const numDeleted = await tx(UserWork.table)
+    .where({ job_id: jobID })
+    .del();
+  return numDeleted;
+}
+
+/**
+ * Deletes all of the rows for the given job from the user_work table.
```

Comment:
Looks like you copied this comment from the function above. It does not mention `serviceID`.

Reply:
Fixed